src/coreclr/vm/interpexec.cpp

```diff
@@ -49,263 +53,267 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
 MAIN_LOOP:
     while (true)
     {
-        // Interpreter-TODO: This is only needed to enable SOS see the exact location in the interpreted method.
-        // Neither the GC nor the managed debugger needs that as they walk the stack when the runtime is suspended
-        // and we can save the IP to the frame at the suspension time.
-        // It will be useful for testing e.g. the debug info at various locations in the current method, so let's
-        // keep it for such purposes until we don't need it anymore.
-        pFrame->ip = (int32_t*)ip;
-
-        switch (*ip)
+        try
         {
+            INSTALL_MANAGED_EXCEPTION_DISPATCHER;
+            INSTALL_UNWIND_AND_CONTINUE_HANDLER;
```

Comment:
I don't think this code is meant to be run for each opcode executed, we could at least extract it out of the loop ? Any reason why this couldn't be a simple try/catch in the future, so we don't have any overhead at all on the non-exceptional path ?